pkg/network/proxy/proxy.go

```diff
@@ -318,36 +329,53 @@ func (proxy *OsdnProxy) OnEndpointsAdd(ep *corev1.Endpoints) {
 	defer proxy.Unlock()
 
 	ns := proxy.getNamespace(ep.Namespace)
-	pep := &proxyEndpoints{ep, proxy.endpointsBlocked(ns, ep)}
-	ns.endpoints[ep.UID] = pep
-	if !pep.blocked {
-		proxy.baseProxy.OnEndpointsAdd(ep)
+	if proxy.endpointsBlockable(ns, ep) {
+		pep := &proxyEndpoints{ep, proxy.endpointsBlocked(ns, ep)}
+		ns.blockableEndpoints[ep.UID] = pep
+		if pep.blocked {
+			return
+		}
 	}
+
+	proxy.baseProxy.OnEndpointsAdd(ep)
 }
 
 func (proxy *OsdnProxy) OnEndpointsUpdate(old, ep *corev1.Endpoints) {
 	proxy.Lock()
 	defer proxy.Unlock()
 
 	ns := proxy.getNamespace(ep.Namespace)
-	pep := ns.endpoints[ep.UID]
+	isBlockable := proxy.endpointsBlockable(ns, ep)
+	isBlocked := isBlockable && proxy.endpointsBlocked(ns, ep)
+
+	pep := ns.blockableEndpoints[ep.UID]
 	if pep == nil {
-		klog.Warningf("Got OnEndpointsUpdate for unknown Endpoints %#v", ep)
```

Comment:
we should keep warning that this endpoint wasn't "blockable" before

Reply:
That was a "can't happen" situation, and with the new code organization, we can't tell the difference between "we don't know about this namespace because it wasn't worth paying attention to before" and "we don't know about this namespace because the informer is buggy". So we just assume it's the former.